crates/ark/src/modules/positron/html_widgets.R

```diff
@@ -9,28 +9,49 @@
     # Render the widget to a tag list.
     rendered <- htmltools::as.tags(x, standalone = TRUE)
 
-    # Resolve the dependencies for the widget (scripts, stylesheets, etc.).
-    dependencies <- htmltools::resolveDependencies(
-        attr(rendered, "html_dependencies", exact = TRUE))
+    # Render the tag list to a temporary file using html_print. Don't view the
+    # file yet; we'll do that in a bit.
+    tmp_file <- htmltools::html_print(rendered, viewer = NULL)
+
+    # Derive the height of the viewer pane from the sizing policy of the widget.
+    is_plot <- x$sizingPolicy$knitr$figure
+    height <- x$sizingPolicy$viewer$paneHeight
+    if (identical(height, 'maximize')) {
+        height <- -1
+    } else if (is.null(height)) {
+        height <- 0
+    }
+
+    # Attempt to derive a label for the widget from its class. If the class is
+    # empty, use a default label.
+    label <- class(x)[1]
+    if (nzchar(label)) {
+        label <- paste(label, "HTML widget")
+    } else {
+        label <- "R HTML widget"
+    }
 
     # Pass the widget to the viewer. Positron will assemble the final HTML
     # document from these components.
-    .ps.Call("ps_html_widget",
-        class(x)[1],
-        list(
-            tags = rendered,
-            dependencies = dependencies,
-            sizing_policy = x$sizingPolicy))
+    .ps.Call("ps_html_viewer",
+        tmp_file,
+        label,
+        height,
+        is_plot)
 }
 
 #' @export
 .ps.viewer.addOverrides <- function() {
     add_s3_override("print.htmlwidget", .ps.view_html_widget)
+    add_s3_override("print.shiny.tag", .ps.view_html_widget)
+    add_s3_override("print.shiny.tag.list", .ps.view_html_widget)
 }
 
 #' @export
 .ps.viewer.removeOverrides <- function() {
     remove_s3_override("print.htmlwidget")
+    remove_s3_override("print.shiny.tag", .ps.view_html_widget)
+    remove_s3_override("print.shiny.tag.list", .ps.view_html_widget)
```

Comment:
```suggestion
    remove_s3_override("print.shiny.tag")
    remove_s3_override("print.shiny.tag.list")
```

I imagine the fact that this slipped through suggests that we haven't actually exercised this code yet, otherwise we would have seen an error already?

Reply:
Yes, this code was adapted from RStudio, where we do this S3 override enable/disable constantly (as a preamble and then cleanup every time you run code in notebook/inline output mode). In Positron we flip the switch on but don't currently turn it off. Felt weird to not have the off switch but maybe I should remove it since we aren't using it...